scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+		lastXY = newPoint;
+	}
+	else
+	{
+		//qDebug() << "movetopoint failed";
+	}
+
+	return pass;
+}
+
+//TODO:, extract some font heights instesad of using dx all the time
+TextRegion::FrameworkLineTests TextRegion::addGlyphAtPoint(QPointF newGlyphPoint, PdfGlyph newGlyph)
+{
+	QPointF movedGlyphPoint = QPointF(newGlyphPoint.x() + newGlyph.dx, newGlyphPoint.y() + newGlyph.dy);
+	//qDebug() << "addGlyphAtPoint start" << newGlyphPoint << " glyph:"<< new_glyph.code;
+	//FIXME: There should be no need for testing the scope when adding a new glyph because moveto should have been called first but leave it in for now to catch any errors in the logic
+	if (glyphs.size() == 1)
+	{
+		// FIXME: do a propper lookup of the height
+		lineSpacing = newGlyph.dx * 2;
+		//qDebug() << "addGlyphAtPoint start";
+		lastXY = newGlyphPoint;
+		lineBaseXY = newGlyphPoint;
+	}	
+	bool xInLimits = false;
+	if (closeToX(newGlyphPoint.x(), lastXY.x()))
+	{
+		xInLimits = true;
+	}
+	bool yInLimits = false;
+	if (closeToY(newGlyphPoint.y(), lastXY.y()))
+	{
+		yInLimits = true;
+	}
+	FrameworkLineTests pass = linearTest(newGlyphPoint, xInLimits, yInLimits);
+	if (pass != FrameworkLineTests::FAIL)
+	{
+		maxHeight = abs(textRegioBasenOrigin.y() - movedGlyphPoint.y()) + lineSpacing > maxHeight ? abs(textRegioBasenOrigin.y() - movedGlyphPoint.y()) + lineSpacing : maxHeight;
+		//moveto deals with setting newlines and segments, all we hve to do is populatit them with the parameters tyhe glyph gives us such as it's width and height and set the glyph index for the newlines and segments
+
+		TextRegionLine* textRegionLine = &textRegionLines.back();
+		if (pass == FrameworkLineTests::NEWLINE || pass == FrameworkLineTests::FIRSTPOINT) {
+			textRegionLine->glyphIndex = glyphs.size() - 1;
+			textRegionLine->baseOrigin = QPointF(textRegioBasenOrigin.x(), newGlyphPoint.y());
+		}
+
+		TextRegionLine *segment = &textRegionLine->segments.back();
+		segment->width = abs(movedGlyphPoint.x() - segment->baseOrigin.x());
+		segment->glyphIndex = glyphs.size() - 1;
+		qreal thisHeight = 0;
+		if (textRegionLines.size() > 1)
+		{
+			thisHeight = abs(newGlyphPoint.y() - textRegionLines[textRegionLines.size() - 2].baseOrigin.y());
+		}
+		else
+		{
+			thisHeight = newGlyph.dx;
+		}
+		segment->maxHeight = thisHeight > segment->maxHeight ? thisHeight : segment->maxHeight;
+		textRegionLine->maxHeight = textRegionLine->maxHeight > thisHeight ? textRegionLine->maxHeight : thisHeight;
+		textRegionLine->width = abs(movedGlyphPoint.x() - textRegionLine->baseOrigin.x());
+
+		maxWidth = textRegionLine->width > maxWidth ? textRegionLine->width : maxWidth;
+		if (textRegionLine->segments.size() == 1)
+		{
+			lineBaseXY = textRegionLine->baseOrigin;
+		}
+		lastXY = movedGlyphPoint;
+	}
+	else
+	{
+		qDebug() << "addGlyphAtPoint failed, this should never happen";
+	}
+	return pass;
+}
+
+void TextRegion::renderToTextFrame(PageItem* textNode, ParagraphStyle& pStyle)

```

Comment:
`addToPageItem()` ?

do you plan to do something with `pStyle`... comment it out or add a todo that tells what the usage will be...

Reply:
ok